analyzer/analyzer.go

```diff
@@ -10,32 +10,40 @@ import (
 	"github.com/manuelarte/embeddedstructfieldcheck/internal"
 )
 
-const ForbidMutexName = "forbid-mutex"
+const (
+	EmptyLineCheck   = "empty-line"
+	ForbidMutexCheck = "forbid-mutex"
+)
 
 func NewAnalyzer() *analysis.Analyzer {
-	var forbidMutex bool
+	var (
+		emptyLine   bool
+		forbidMutex bool
+	)
 
 	a := &analysis.Analyzer{
 		Name: "embeddedstructfieldcheck",
 		Doc: "Embedded types should be at the top of the field list of a struct, " +
 			"and there must be an empty line separating embedded fields from regular fields.",
 		URL: "https://github.com/manuelarte/embeddedstructfieldcheck",
 		Run: func(pass *analysis.Pass) (any, error) {
-			run(pass, forbidMutex)
+			run(pass, emptyLine, forbidMutex)
 
 			//nolint:nilnil // impossible case.
 			return nil, nil
 		},
 		Requires: []*analysis.Analyzer{inspect.Analyzer},
 	}
 
-	a.Flags.BoolVar(&forbidMutex, ForbidMutexName, false,
+	a.Flags.BoolVar(&emptyLine, EmptyLineCheck, true,
+		"Checks that there is an empty space between the embedded fields and regular fields.")
+	a.Flags.BoolVar(&forbidMutex, ForbidMutexCheck, false,
 		"Checks that sync.Mutex and sync.RWMutex are not used as an embedded fields.")
 
 	return a
 }
 
-func run(pass *analysis.Pass, forbidMutex bool) {
+func run(pass *analysis.Pass, emptyLine, forbidMutex bool) {
```

Comment:
I would pass a struct here.


```suggestion
type checkerOptions struct {
  emptyLine    bool
  forbidMutex bool
}

func run(pass *analysis.Pass, opts checkerOptions) {
```


It's open to debate if it doesn't sound to a too-early optimization.

Here having 2 fields is OK, but if you plan to add other options it could be considered

Reply:
Clearly early optimization.

Reply:
This linter is checking three things:
- Embedded fields declared before regular fields.
- Empty line separating embedded fields and regular fields.
- sync.Mutex and sync.RWMutex are not embedded.

So far, there are two 2/3 settings, so I think it can't grow much more.

But this is something to take into account in case we want to increase the functionality provided by this linter.